build.rs

```diff
@@ -40,5 +48,60 @@ fn main() {
         println!("cargo:rustc-link-lib=static=chck-xdg");
         println!("cargo:rustc-link-lib=static=wlc-protos");
         println!("cargo:rustc-link-lib=static=wlc");
+    } else if let Err(s) = find() {
+        let _ = writeln!(io::stderr(), "{}", s);
+        process::exit(1);
+    }
+}
+
+fn find() -> Result<(), Error> {
+
+    let package_name = "wlc";
+    let shared_libs = ["wlc"];
+
+    // https://github.com/way-cooler/rust-wlc#rust-wlc currently
+    // (i.e. commit 97e0d5816fe7ec97dae20196d1e25c550f60f08f) suggests
+    // version 0.0.7.
+    let version = if cfg!(feature = "0.0.8") {
+        "0.0.8"
+    } else if cfg!(feature = "0.0.7") {
+        "0.0.7"
+    } else if cfg!(feature = "0.0.6") {
+        "0.0.6"
+    } else {
+        "0.0.5"
+    };
+
+    // Is this Windoze-specific? Do we need to carry this along on Linux with
+    // Wayland at all?
+    let target = env::var("TARGET").unwrap();
+    let hardcode_shared_libs = target.contains("windows");
+
+    let mut config = Config::new();
+    config.atleast_version(version);
+
+    if hardcode_shared_libs {
+        config.cargo_metadata(false);
+    }
+
+    match config.probe(package_name) {
+        Ok(library) => {
+            if hardcode_shared_libs {
+                for lib_ in shared_libs.iter() {
+                    println!("cargo:rustc-link-lib=dylib={}", lib_);
+                }
+                for path in library.link_paths.iter() {
+                    println!("cargo:rustc-link-search=native={}", path.to_str().unwrap());
+                }
```

Comment:
To be honest, I am a little confused as to why `rust-wlc` would need to specify all these libraries
explicitly.  The linker should be able to figure those out recursively when it reads those libraries
that can be obtained by `pkg-config --libs --cflags wlc`.

BTW, in what build scenarios do you envision the `CARGO_FEATURE_STATIC_WLC`
environment variable to be specified?